# HG changeset patch
# User twisti
# Date 1361822075 28800
# Node ID cf2b7f0f7a2188a1421bdd033e88bf4b0aa6870c
# Parent  f9d5be4340c0ab09bebfcd2dce5563a2042ccc95
8004336: Better handling of method handle intrinsic frames
Reviewed-by: kvn, jrose, ahgross

diff --git a/src/share/vm/opto/library_call.cpp b/src/share/vm/opto/library_call.cpp
--- hotspot/src/share/vm/opto/library_call.cpp
+++ hotspot/src/share/vm/opto/library_call.cpp
@@ -1313,7 +1313,6 @@
   push(result);
   return true;
 }
-
 //--------------------------pop_math_arg--------------------------------
 // Pop a double argument to a math function from the stack
 // rounding it if necessary.
@@ -3746,7 +3745,8 @@
       }
     }
   }
-  else if (method->is_method_handle_adapter()) {
+
+  if (method->is_method_handle_adapter()) {
     // This is an internal adapter frame from the MethodHandleCompiler -- skip it
     return true;
   }
